langchain/tests/test_retrievers.py

```diff
@@ -197,24 +27,10 @@ def test_retrieve_invoke_args(
     pipeline_id: str,
 ) -> None:
     retriever = VectorizeRetriever(environment=environment, api_token=api_token)
-    start = time.time()
-    while True:
-        try:
-            docs = retriever.invoke(
-                input="What are you?",
-                organization=org_id,
-                pipeline_id=pipeline_id,
-                num_results=2,
-            )
-            if len(docs) == 2:
-                break
-
-        except Exception as e:
-            if "503" in str(e):
-                continue
-            raise RuntimeError(e) from e
-        if time.time() - start > 180:
-            msg = "Docs not retrieved in time"
-            raise RuntimeError(msg)
-
-        time.sleep(1)
+    docs = retriever.invoke(
```

Comment:
we should keep the retries

Reply:
The idea is to have the retries in the setup (in the pipeline_id fixture).
Is there a possibility that retrieving docs fails after it has succeeded once ?

Reply:
No, in that case it's a different issue and we should fail